src/bin/conserve.rs

```diff
@@ -372,16 +385,24 @@ fn list_tree_contents<T: ReadTree>(tree: &T) -> Result<()> {
 
 fn restore(subm: &ArgMatches) -> Result<()> {
     let dest = Path::new(subm.value_of("destination").unwrap());
+    // let only_path = Path::new(subm.value_of("only").unwrap());
+    let only_path = Path::new(subm.value_of("only").unwrap_or(""));
```

Comment:
This looks like it takes only a single path?

Reply:
It seems there's a bit of divergence in this patch about whether it takes a single path, or several, and whether it's a glob or a path.

Overall, I think it makes sense to have two mechanisms:

1. "Negative excludes", like in [gitignore](https://git-scm.com/docs/gitignore), that allow files that would otherwise be excluded. This should go into the exclusion mechanism and then can apply to backup, restore, and others. There can be several of these and they can be ordered with regard to other exclusions. These need to be matched against every entry apath. These seem to need some care not exclude the parents of the thing you want.

2. Restore of a subtree, which is what I'd expect `--only` to do. Because this takes just one path, it can be smarter about seeking to that point in the index without reading everything, and it can either include the parents of that path, or strip them off, like we discussed in #112. 

This code seems a little inconsistent at present which one it's trying to do, but it seems like it's mostly the second, so I'd suggest to just update the option to take only one value and say it's a path, not a string.

Reply:
This takes a single path indeed. Once user provides that path to be included, all its subpaths will be included too and restored at the same relative path in the destination.
It's true that the current implementation needs to read everything and skip what is not in the `only` path.

The current option takes one value already, or am i missing something?
```    
fn include_only_arg<'a, 'b>() -> Arg<'a, 'b> {
        Arg::with_name("only")
            .long("only")
            .short("i")
            .takes_value(true)
            .multiple(true)
            .number_of_values(1)
            .required(false)
            .help("Include only files that match the provided glob pattern")
    }
```

Reply:
I'm not sure why you're using `multiple(true)` there? https://docs.rs/clap/2.33.1/clap/struct.Arg.html#method.multiple

And, the help text says it's a glob.